### PR TITLE
Experimental/oneof combinator

### PR DESCRIPTION
**Pull request by ascandone**

### CODE REVIEW

File: internal/parser/antlr/numscript_parser.go [hunk 1]
```diff
@@ -2763,6 +2769,85 @@
...
+		listenerT.EnterSrcOneof(s)
+	}
+}
+
+func (s *SrcOneofContext) ExitRule(listener antlr.ParseTreeListener) {
+	if listenerT, ok := listener.(NumscriptListener); ok {
+		listenerT.ExitSrcOneof(s)
+	}
+}
+
```
[the remark this answers is not in the document]
Comment (ascandone) [reply internal/parser/antlr/numscript_parser.go:2850]: that's autogenerated code (as you can notice from the header)
please don't bother me about machine-generated code and its test coverage

File: internal/interpreter/interpreter.go [hunk 2]
```diff
@@ -476,6 +476,16 @@
...
+	case *parser.SourceOneof:
+		err := s.checkFeatureFlag(ExperimentalOneofFeatureFlag)
+		if err != nil {
+			return nil, err
+		}
+
+		// we can safely access the first one because empty oneof is parsing err
+		first := source.Sources[0]
+		return s.sendAll(first)
+
```
[the remark this answers is not in the document]
Comment (ascandone) [reply internal/interpreter/interpreter.go:488]: that's not inconsistent: this is indeed a backtracking operator, but when calling this method none of the branch can fail, hence we just forward to the first source

File: internal/parser/parser_test.go [hunk 3]
```diff
@@ -103,6 +112,19 @@
...
+	source = @s
+	destination = oneof {
+		max $m1 to @d1
+		max $m2 kept
+		remaining to @d3
+  }
+	)`)
+	require.Empty(t, p.Errors)
+	snaps.MatchSnapshot(t, p.Value)
+}
```
[the remark this answers is not in the document]
Comment (ascandone) [reply internal/parser/parser_test.go:126]: these edge cases are already taken care of in the inorder parsing fault tolerance tests

**Comment by paul-nicolas** [comment]
Is it possible to follow the conventional commit described here please : https://www.conventionalcommits.org/en/v1.0.0 ?

**Review by paul-nicolas — APPROVED** [review]
LGTM
